BUILDING.md

```diff
@@ -162,3 +166,101 @@ publishing {
         ...
 ```
 Use `./gradlew publishShadowPublicationToStagingRepository` to produce maven artifacts. When building as part of the monolithic distribution customize `scripts/build.sh` to collect maven artifacts. See [job-scheduler#71](https://github.com/opensearch-project/job-scheduler/pull/82/files) for an example.
+
+## Custom Gradle Plugins
+
+[OpenSearch](https://github.com/opensearch-project/OpenSearch/tree/main/buildSrc) provides a set of custom gradle plugins that avoid boilerplate code for building and testing plugins.
+
+### opensearch.pluginzip
+
+This plugin identifies the generated zip file distribution which is the ouput of `bundlePlugin` task and publishes to local maven repo with standardized maven coordinates.
+[Plugin Code](https://github.com/opensearch-project/OpenSearch/tree/main/buildSrc/src/main/java/org/opensearch/gradle/pluginzip), [Plugin Tests](https://github.com/opensearch-project/OpenSearch/tree/main/buildSrc/src/test/java/org/opensearch/gradle/pluginzip), [Plugin META-INF](https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/main/resources/META-INF/gradle-plugins/opensearch.pluginzip.properties)
+
+
+#### Plugin Design
+
+* `opensearch.pluginzip` is java based code published into build-tools.
+* The maven coordinates groupID is fixed as `org.openserach.plugin`, `version` and `artifcatID` will be inferred from gradle project properties.
+* User can also pass custom POM extensions, that will add desired xml to the zip maven POM file geneaterd during runtime.
+* Once the plugin is added to the `build.gradle` as `apply plugin: 'opensearch.pluginzip'`, this will add a new custom publish task `publishPluginZipPublicationToZipStagingRepository`, this task will publish the zip distribution to the maven local (file system).
+* The maven local artifacts can then be published to maven central/nexus.
+* The plugin will not add jars generated by tasks `sourcesJar` and `javadocJar`, this is done in purpose to exclude `jars` for `zip` publications.
+
+#### Plugin Usage
+
+1. The Key requirement for this plugin to work is that the zip should be generated by `bundlePlugin` task that comes from existing [PublishPlugin](https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/main/java/org/opensearch/gradle/PublishPlugin.java). 
```

Comment:
Nit: try to shorten things, e.g. "The Key requirement for this plugin to work is that the zip should be generated by `bundlePlugin` task" is the same as "Ensure that your project uses bundlePlugin".

Reply:
Fixed this @dblock